### PR TITLE
Add an example of Lime text explainer from AIX360 toolkit

### PR DESCRIPTION
**Pull request by C1berwiz**
**What this PR does / why we need it**:
This PR added an example of LIME text explainer of AIX360 toolkit
Since there are 3 modes of LIME explainer (image, text and tabular ), but currently there is only a sample of image (MNIST)
We suppose that adding a text example would help others better understanding LIME explainer

**Type of changes**
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] Test A
- like the mnist sample of aix and mentioned in README.md , this sample should execute just like mnist sample
- The first step is to [determine the ingress IP and ports](../../../../../README.md#determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`

```
MODEL_NAME=aix-explainer
SERVICE_HOSTNAME=$(kubectl get inferenceservice ${MODEL_NAME} -o jsonpath='{.status.url}' | cut -d "/" -f 3)
python query_explain.py http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/${MODEL_NAME}:explain ${SERVICE_HOSTNAME}
```
Then you will see the explanation result as several charts. 

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yuzisun** [comment]
@drewbutlerbb4 Can you help review this?

**Comment by Tomcli** [comment]
@C1berwiz I'm not sure do we want to host a big binary file like [rfmodel.pickle](https://github.com/kserve/kserve/pull/2355/files#diff-cecaba4f0caa00d6cbe43637c9043bc9ce59a4bd08c466e9977f85d21a239a31) in the KServe repo. Usually we either provide a simple script to train the model or ask the KServe community to upload it to the community GCS bucket.

@yuzisun do you know a good process for uploading binary as a contributor?

**Comment by yuzisun** [comment]
> @C1berwiz I'm not sure do we want to host a big binary file like [rfmodel.pickle](https://github.com/kserve/kserve/pull/2355/files#diff-cecaba4f0caa00d6cbe43637c9043bc9ce59a4bd08c466e9977f85d21a239a31) in the KServe repo. Usually we either provide a simple script to train the model or ask the KServe community to upload it to the community GCS bucket.
> 
> @yuzisun do you know a good process for uploading binary as a contributor?

We can help upload the model to community GCS bucket.

**Comment by yuzisun** [comment]
@C1berwiz Can you help sign the commit?

**Comment by C1berwiz** [comment]
@yuzisun I've signed the commits, thanks

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/rfserver/model.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self._step_func = step_func
+
+    def fit(self, *args):
+        return self
+
+    def transform(self, X):
+        return self._step_func(X)
+
+
+class RFModel(kserve.Model):  # pylint:disable=c-extension-no-member
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/rfserver/model.py]: Any reason why we need to implement this? We can use the out of the box sklearn server directly to serve the model.

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Using AIX to get explanations for NewsGroups classifications
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/README.md:1]: Let's only leave the code examples here in kserve repo and move the documentation to website [explainer folder](https://github.com/kserve/website/tree/main/docs/modelserving/explainer/aix)

File: docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    def fit(self, *args):
+        return self
+
+    def transform(self, X):
+        return self._step_func(X)
+
+
+path_to_save = sys.argv[1]
+
+dataset = sys.argv[2]
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py]: Do we need this line?

**Comment by yuzisun** [comment]
> > @C1berwiz I'm not sure do we want to host a big binary file like [rfmodel.pickle](https://github.com/kserve/kserve/pull/2355/files#diff-cecaba4f0caa00d6cbe43637c9043bc9ce59a4bd08c466e9977f85d21a239a31) in the KServe repo. Usually we either provide a simple script to train the model or ask the KServe community to upload it to the community GCS bucket.
> > @yuzisun do you know a good process for uploading binary as a contributor?
> 
> We can help upload the model to community GCS bucket.

@C1berwiz The model has been uploaded to gs://kfserving-examples/models/sklearn/newsgroup/model.pickle

File: docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py (hunk 3 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py]: @yuzisun The line was used for testing other dataset
we forgot to remove that, we will fix that later

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md (hunk 2 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/README.md:1]: Ok we'll update the docs to there

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/rfserver/model.py (hunk 1 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/rfserver/model.py]: @yuzisun sorry for letting you wait for so long
We are now using out of the box model 
thanks

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+This is another explanation example for text compare to [MNIST classifications (for image)](https://github.com/kserve/kserve/tree/master/docs/samples/explanation/aix/mnist).
+
+To deploy the inferenceservice with v1beta1 API
+
+`kubectl apply -f aix-explainer.yaml`
+
+Then find the url.
+
+`kubectl get inferenceservice`
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/README.md]:
```suggestion
`kubectl get inferenceservice aix-explainer`
```

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+```
+NAME         URL                                               READY   DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
+aixserver   http://aixserver.somecluster/v1/models/aixserver   True    100                                40m
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/README.md]: seems like the name does not match here

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+# limitations under the License.
+
+from setuptools import setup, find_packages
+
+tests_require = [
+    'pytest',
+    'pytest-tornasync',
+    'mypy'
+]
+setup(
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py]: Do we still want this setup file as the rfserver is no longer needed ?

File: python/aixexplainer/aixserver/model.py [hunk 7]
```diff
@@ -102,6 +110,16 @@ def explain(self, request: Dict) -> Dict:
                     "masks": masks,
                     "top_labels": np.array(explanation.top_labels).astype(np.int32).tolist()
                 }}
+            elif str.lower(self.explainer_type) == "limetexts":
```
Comment (yuzisun) [inline python/aixexplainer/aixserver/model.py:113]: Can we add an e2e test for limetexts like the [limeimage test](https://github.com/kserve/kserve/blob/master/test/e2e/explainer/test_aix_explainer.py)?

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md (hunk 4 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/README.md]: @yuzisun I've signed-off and commit suggestion, thanks.

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+# Random Forest for NewsGroups on KServe
```
Comment (yuzisun) [inline docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/README.md]: We can delete this file now

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py (hunk 6 above)
Comment (yuzisun) [reply docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py]: Let's delete this file

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/README.md (hunk 8 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/README.md]: @yuzisun I've deleted the file, thanks

File: docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py (hunk 6 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/rfserver/setup.py]: @yuzisun I've deleted the file, thanks

File: docs/samples/explanation/aix/Fetch_20newsgroups/README.md (hunk 5 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/README.md]: @yuzisun Updated ,thanks

File: docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py (hunk 3 above)
Comment (C1berwiz) [reply docs/samples/explanation/aix/Fetch_20newsgroups/train_rf.py]: @yuzisun that line has been deleted

**Comment by yuzisun** [comment]
Thanks @C1berwiz for the awesome contribution !! 

Can you help move the example to the website under aix explainer?https://github.com/kserve/website/tree/main/docs/modelserving/explainer/aix/mnist

/lgtm 
/approve

**Pull request merged**